src/System.Threading.Tasks.Channels/System/IO/Channels/BoundedChannelFullMode.cs

```diff
@@ -2,7 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-namespace System.Threading.Tasks.Channels
+namespace System.IO.Channels
 {
     /// <summary>Specifies the behavior to use when writing to a bounded channel that is already full.</summary>
     public enum BoundedChannelFullMode
```

Comment:
The enum name is strange. I am not sure if we discussed it at the API review, but maybe:
1. We nest the enum in the type
2. We call the enum OverflowOptions
thoughts?

Reply:
Also, is there a channle or mode where writers can get a callback when the channel overflows?

Reply:
> I am not sure if we discussed it at the API review

We discussed it, but didn't come to any conclusions about better names.

> We nest the enum in the type

In which type?  BoundedChannelOptions?

> We call the enum OverflowOptions

I see the appeal, but personally that suggests to me it's about responding to an overflow having already occurred, whereas this is about what happens when one would occur if we didn't take action.

Reply:
Yeah, I was thinking about nested in BoundedChannelOptions.

Reply:
I've been playing around with it.  Nesting it ends up being odd, as then the name you want for the type ends up being the same as the name you want for the property, but you can't name them the same.  I also think "full mode" is a fine description.  I'm going to leave it as-is for now; we can choose to rename it / reorganize it if there's feedback to that effect before it ships.